src/sfpowerkitCommand.ts

```diff
@@ -0,0 +1,82 @@
+import { SfdxCommand } from "@salesforce/command";
+import SFPKLogger, {COLOR_HEADER, LoggerLevel} from "./SFPKLogger";
+import { SFPowerkit } from "./sfpowerkit";
+
+/**
+ * A base class that provides common funtionality for sfpowerscripts commands
+ *
+ * @extends SfdxCommand
+ */
+export default abstract class SFPowerkitCommand extends SfdxCommand {
+  public static isJsonFormatEnabled: boolean;
+  public static logLevel;
+  private sfpowerkitConfig;
+
+  /**
+   * Command run code goes here
+   */
+  abstract execute(): Promise<any>;
+
+  /**
+   * Entry point for the commands
+   */
+  async run(): Promise<any> {
+    SFPowerkit.setLogLevel(this.flags.loglevel, this.flags.json);
+
+    // Always enable color by default
+    if (process.env.SFPOWERKIT_NOCOLOR) SFPowerkit.disableColor();
+    else SFPowerkit.enableColor();
+
+    this.setLogLevel();
+
+    for (const plugin of this.config.plugins) {
+      if (plugin.name === "sfpowerkit") {
+        this.sfpowerkitConfig = plugin;
+        }
+      }
+
+    if (!this.flags.json) {
+      this.sfpowerkitHeader();
+    }
+    
+    return this.execute();
+  }
+
+  private sfpowerkitHeader() {
+    if (!process.env.SFPOWERKIT_NOHEADER) {
+    SFPKLogger.log(
+      COLOR_HEADER(
+        `-------------------------------------------------------------------------------------------`
+      )
+    );
+    SFPKLogger.log(
```

Comment:
As mentioned in previous comment, use `Sfpowerkit.log` instead of `SFPKLogger.log` until you are ready to use `SFPLogger` throughout the codebase.